src/features/duckplayer/overlay-messages.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable promise/prefer-await-to-then */
+import * as constants from './constants.js'
+
+/**
+ * @typedef {import("@duckduckgo/messaging").Messaging} Messaging
+ *
+ * A wrapper for all communications.
+ *
+ * Please see https://duckduckgo.github.io/content-scope-utils/modules/Webkit_Messaging for the underlying
+ * messaging primitives.
+ */
+export class DuckPlayerOverlayMessages {
+    /**
+     * @param {Messaging} messaging
+     * @internal
+     */
+    constructor (messaging) {
+        /**
+         * @internal
+         */
+        this.messaging = messaging
+    }
+
+    /**
+     * Inform the native layer that an interaction occurred
+     * @param {import("../duck-player.js").UserValues} userValues
+     * @returns {Promise<import("../duck-player.js").UserValues>}
+     */
+    setUserValues (userValues) {
+        return this.messaging.request(constants.MSG_NAME_SET_VALUES, userValues)
+    }
+
+    /**
+     * @returns {Promise<import("../duck-player.js").UserValues>}
+     */
+    getUserValues () {
+        return this.messaging.request(constants.MSG_NAME_READ_VALUES, {})
+    }
+
+    /**
+     * @param {Pixel} pixel
+     */
+    sendPixel (pixel) {
+        this.messaging.notify(constants.MSG_NAME_PIXEL, {
```

Comment:
Remember that once we consider rolling this for extensions we need to ensure that this is disabled for Firefox.

We should probably fast follow to ensure that firefox isn't enable-able even if this overlays file is bundled (which I know it's not right now)